sub/sd_ass.c

```diff
@@ -508,7 +508,7 @@ static void configure_ass(struct sd *sd, struct mp_osd_res *dim,
     if (converted || shared_opts->ass_style_override[sd->order] == ASS_STYLE_OVERRIDE_FORCE) {
         set_scale_with_window = opts->sub_scale_with_window;
         set_use_margins = opts->sub_use_margins;
-        set_scale_by_window = opts->sub_scale_by_window;
+        set_scale_by_window = opts->sub_scale_by_window && dim->h > 720;
```

Comment:
Sorry to instantly start another bikeshed, but do we really want to do this for subtitles? Of course, small subtitles that you can't read are useless but having overly large ones that completely cover up the picture is maybe worse?

Reply:
I agree. scale-by-window should also scale down by default. Otherwise, for smaller window the subs can end up too big. scale-by-window is not scale-only-up-by-window (and it's not obvious up from what. 720 as the threshold seems arbitrary from a user point of view).

Reply:
I thought we settled on leaving the scaling enabled by default with https://github.com/mpv-player/mpv/pull/14158#issuecomment-2120732672. I think it would be inconsistent for the OSD and subs to have different default scaling behavior.

Reply:
To me they seem like different usecases. For builtin/OSD stuff, the user actively opts into those some way (showing stats, osc, etc.) so them overly obscuring the underlying video is a given. subtitles are more passive to me and ideally should be unobtrusive. Of course some overly complicated ass subtitles disagree with me but that's a different story.

I realize that trying to optimize the UI of a small mpv window is kind of hard though.

Reply:
¯\\_(ツ)_/¯ I don't like the scaling either but kasper's point that it keeps text including subtitles readable on large screens several meters away makes sense and both VLC and MPC-HC scale subtitles.

Reply:
If you have a use case (with specific window sizes) which this commit improves, then the commit message should mention how it behaves before and after this commit.

You simply can't optimize text size for a small window, so the best you can do is to scale it down and if it's unreadable then so be it.

Otherwise with a small window (think PIP-like mode) the text canj and will definitely be too big.

Reply:
I'm not sure I understand how the 720p cutoff helps? I understand that with very small window sizes certain elements become "too small", but it is already the case for OSC, subtitles, stats and so on... and most importantly video. If someone using 100x100 window, it is already too late for them. I would argue it is better to keep relative size of everything as if it were program thumbnail.

I understand what you are trying to do to make things readable at lower window sizes, but is this really a problem? Also note that at 100x100 you won't fit any meaningful amount of subtitle text anyway.